src/rgw/rgw_admin.cc

```diff
@@ -1363,32 +1363,46 @@ int do_check_object_locator(const string& tenant_name, const string& bucket_name
   return 0;
 }
 
-#define MAX_REST_RESPONSE (128 * 1024) // we expect a very small response
-static int send_to_remote_gateway(const string& remote, req_info& info,
-                                  bufferlist& in_data, JSONParser& parser)
+/// search for a matching zone/zonegroup id and return a connection if found
+static boost::optional<RGWRESTConn> get_remote_conn(RGWRados *store,
+                                                    const RGWZoneGroup& zonegroup,
+                                                    const std::string& remote)
```

Comment:
This causes a build failure on CentOS 7.3 and Ubuntu Trusty, which presumably have too-old boost versions:

```
rgw/rgw_admin.cc: In function 'boost::optional<RGWRESTConn> get_remote_conn(RGWRados*, const RGWZoneGroup&, const string&)':
rgw/rgw_admin.cc:1373:10: error: 'class boost::optional<RGWRESTConn>' has no member named 'emplace'
     conn.emplace(store->ctx(), store, remote, zonegroup.endpoints);
          ^
```